### ISSUE TITLE
file module stop when file doesn't belong to an existing user

### ISSUE THREAD

**Issue by bleader** [ansible/ansible#932]
When calling the file module to fixup the owner of a folder,  currently belonging to an unexisting uid:

```
ls -l /path/test
drwxr-xr-x   2        7779 users           4096 Jun 22 09:57 test
```

called like this:

```
ansible host -m file -a "path=/path/test owner=test group=test state=directory"
```

I end up with this Traceback:

```
host | FAILED => failed to parse: Traceback (most recent call last):

  File "/var/tmp/ansible.Wkcuu9/file", line 321, in <module>

    changed = set_owner_if_different(path, owner, changed)

  File "/var/tmp/ansible.Wkcuu9/file", line 217, in set_owner_if_different

    user, group = user_and_group(path)

  File "/var/tmp/ansible.Wkcuu9/file", line 190, in user_and_group

    user = pwd.getpwuid(uid)[0]

KeyError: 'getpwuid(): uid not found: 7779'
```

**Comment by bleader** [comment]
I just realized I was using a quite old clone (2012-06-05), with a pull of devel today, I get:

```
host | success >> {
    "changed": false,
    "group": "users",
    "invocation": {
        "module_args": "path=/path/test user=test groud=test state=directory",
        "module_name": "file"
    },
    "mode": "0755",
    "owner": "7779",
    "path": "/path/test",
    "state": "directory"
}
```

Which is nicer, but still not applying the changes.

**Comment by mpdehaan** [comment]
Didn't say failed though.   What command did you give with the above and which changes does it not seem to be applying?

**Comment by mpdehaan** [comment]
nevermind, I see the owner is still the uuid.   Are you still looking at trying to fix this one yourself possibly?

**Comment by bleader** [comment]
I would like to, but I'm not really at ease with debugging modules yet. I'm going through the documentation.

**Comment by mpdehaan** [comment]
./hacking/test-module -m library/modulename -a "arguments=here"  

and adding some temporary print output is usually a good bet.

**Comment by mpdehaan** [comment]
when specifying the owner and the file was previous unowned I see that it does correctly set the owner to the new account.

If left out, it will tell you what the UID was.

So, looks fine to me.

**Comment by bleader** [comment]
From my tests today, this it was working fine using test-module, but not "remotely" using the same command line in ansible, how did you test it on your side?

**Comment by mpdehaan** [comment]
Doesn't seem like remoteness would enter into the equation.

It may be that your module path was deploying an older version and with
test-module you had picked the one from the checkout?

**Comment by bleader** [comment]
I went through some more test, and understood my mistake, between my two first comments, I upgraded my outdated tree, and didn't carefully looked at the changelog. Mostly, I had no idea the default user wasn't root anymore, thus I was running as a user the ansible command, not specifying the -u parameter.

Obviously for the test-module to work, I had to switch to root as it was directly running the commands without remotely connecting to anything, thus the success of the chown.

As a conclusion, with devel branch, and specifying the -u root, everything works fine. So we can keep it closed as it is, and all forget that I'm stupid! Thanks for your help and your time :)
